src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -4748,6 +4756,7 @@ BoundNode bindElement(CollectionElementSyntax syntax, BindingDiagnosticBag diagn
             {
                 return syntax switch
                 {
+                    ExpressionElementSyntax { Expression: CollectionExpressionSyntax nestedCollectionExpression } => BindCollectionExpression(nestedCollectionExpression, diagnostics, nestingLevel + 1),
```

Comment:
>nestingLevel + 1 [](http://example.com/codeflow?start=183&length=16)

Consider passing in `nestingLevel` as a parameter to avoid a closure class. (It looks like `bindElement()` uses `this` but does not capture any variables currently.)

Reply:
I'll mark these `static` then. I don't want to try and maintain that closure invariant, it's not guaranteed by anything and isn't easy to verify.